src/Fixer/StringNotation/SingleQuoteFixer.php

```diff
@@ -56,14 +73,27 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
 
             if (
                 '"' === $content[0] &&
-                false === strpos($content, "'") &&
+                (false === strpos($content, "'") || true === $this->configuration['strings_containing_single_quote_chars']) &&
                 // regex: odd number of backslashes, not followed by double quote or dollar
                 !preg_match('/(?<!\\\\)(?:\\\\{2})*\\\\(?!["$\\\\])/', $content)
             ) {
                 $content = substr($content, 1, -1);
-                $content = str_replace(['\\"', '\\$'], ['"', '$'], $content);
+                $content = str_replace(['\\"', '\\$', '\''], ['"', '$', '\\\''], $content);
```

Comment:
isn't this a BC break, as it always does this change no matter what has been configured?

Reply:
No because the code enters here only if the configuration is enabled.